drivers/src/sha512.rs

```diff
@@ -0,0 +1,40 @@
+// Licensed under the Apache-2.0 license
+
+use caliptra_error::{CaliptraError, CaliptraResult};
+use caliptra_registers::sha512::Sha512Reg;
+
+use crate::{Array4x12, Array4x8};
+
+pub struct Sha512 {
+    sha512: Sha512Reg,
+}
+
+impl Sha512 {
+    pub fn new(sha512: Sha512Reg) -> Self {
+        Self { sha512 }
+    }
+
+    pub fn gen_pcr_hash(&mut self, nonce: Array4x8) -> CaliptraResult<Array4x12> {
+        let reg = self.sha512.regs_mut();
+
+        let status_reg = reg.gen_pcr_hash_status();
+
+        // Wait for the registers to be ready
+        while !status_reg.read().ready() {}
+
+        // Write the nonce into the register
+        reg.gen_pcr_hash_nonce().write(&nonce.into());
+
+        // Use the start command to start the digesting process
+        reg.gen_pcr_hash_ctrl().write(|ctrl| ctrl.start(true));
```

Comment:
Does mode need to be set first? https://github.com/chipsalliance/caliptra-sw/blob/9f1d077f7ba1e463c23c63b50b19809209c1e74c/drivers/src/sha384.rs#L266